### PR TITLE
feat: [CDS-41932]: artifactory labels

### PR DESCRIPTION
**Pull request by vivekkarnatiharness**
## Describe your changes
This is backend changes for artifactory labels support

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/39911)

### CODE REVIEW

File: 960-api-services/src/main/java/io/harness/artifactory/service/ArtifactoryRegistryServiceImpl.java [hunk 1]
```diff
@@ -114,4 +116,18 @@ private BuildDetailsInternal getBuildNumber(ArtifactoryConfigRequest artifactory
...
+  @Override
+  public List<Map<String, String>> getLabels(
+      ArtifactoryConfigRequest artifactoryConfig, String imageName, String repositoryName, String buildNos) {
+    // Calling getLabels implemented from cg side.
+
+    try {
+      return artifactoryClient.getLabels(artifactoryConfig, imageName, repositoryName, buildNos);
+
+    } catch (Exception e) {
+      // Catching this exception because artifact repository api is returning 404 for some open source images
```
Comment (puthrayaharness) [inline 960-api-services/src/main/java/io/harness/artifactory/service/ArtifactoryRegistryServiceImpl.java:129]: Please log the exception

File: 960-api-services/src/main/java/io/harness/artifactory/service/ArtifactoryRegistryServiceImpl.java [hunk 2]
```diff
@@ -114,4 +116,18 @@ private BuildDetailsInternal getBuildNumber(ArtifactoryConfigRequest artifactory
   public boolean validateCredentials(ArtifactoryConfigRequest artifactoryConfig) {
     return artifactoryClient.validateArtifactServer(artifactoryConfig);
   }
+
+  @Override
+  public List<Map<String, String>> getLabels(
```
Comment (puthrayaharness) [inline 960-api-services/src/main/java/io/harness/artifactory/service/ArtifactoryRegistryServiceImpl.java:121]: Please add unit test case

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/artifacts/artifactory/ArtifactoryArtifactTaskHandler.java [hunk 3]
```diff
@@ -52,22 +53,41 @@ public ArtifactTaskExecutionResponse getLastSuccessfulBuild(
       ArtifactSourceDelegateRequest artifactSourceDelegateRequest) {
     ArtifactoryArtifactDelegateRequest attributesRequest =
         (ArtifactoryArtifactDelegateRequest) artifactSourceDelegateRequest;
+
+    List<Map<String, String>> labels;
```
Comment (puthrayaharness) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/artifacts/artifactory/ArtifactoryArtifactTaskHandler.java:57]: Please use either `labels` or `label`. Please make sure to be consistent across. 
Please make sure we are consistent between Docker, ECR etc.

**Review by puthrayaharness — CHANGES_REQUESTED** [review]

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/artifacts/artifactory/ArtifactoryArtifactTaskHandler.java (hunk 3 above)
Comment (vivekkarnatiharness) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/task/artifacts/artifactory/ArtifactoryArtifactTaskHandler.java:57]: Labels declared here is list of single map.Thats why we have named it as labels, In docker and ecr as well,we are declaring like this

**Comment by vivekkarnatiharness** [comment]
trigger messagecheck

**Comment by harshjain12** [comment]
trigger messagecheck

**Review by puthrayaharness — APPROVED** [review]
> Reviewed 8 of 9 files at r1, 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/39911)* status: :shipit: complete! ✓

**Pull request merged**